homeassistant/components/bitvavo/sensor.py

```diff
@@ -0,0 +1,268 @@
+        self._icon = CURRENCY_ICONS.get(self.unit_of_measurement, DEFAULT_COIN_ICON)
+
+    def _get_data_property(self, property_name):
+        """Return the property from self.coordinator.data."""
+        return self.coordinator.data[CONF_BALANCES][self._balance][property_name]
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        state = float(self._get_data_property("available")) + float(
+            self._get_data_property("inOrder")
+        )
+        return round(state, 4)
+
+    @property
+    def unique_id(self):
+        """Return a unique id for the sensor."""
+        return self._unique_id
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._get_data_property("symbol")
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return self._icon
+
+    @property
+    def extra_state_attributes(self):
+        """Return additional sensor state attributes."""
+        return {
+            "available": self._get_data_property("available"),
+            "in order": self._get_data_property("inOrder"),
+            f"{ASSET_VALUE_BASE}_value".lower(): float(
+                self._get_data_property("asset_value_in_base_asset")
+            ),
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+        }
+
+
+class OpenOrders(CoordinatorEntity):
+    """Implementation of the open orders sensor."""
+
+    def __init__(self, coordinator: BitvavoDataUpdateCoordinator, order):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+
+        self._name = "Bitvavo Open Orders"
+        self._unique_id = "bitvavo_orders_open"
+
+    def _get_orders(self):
+        """Return the data from self.coordinator.data."""
+        return self.coordinator.data[CONF_OPEN_ORDERS]
+
+    def _type_orders(self, side):
+        """Return the number of orders per type."""
+        number = 0
+        for order in self.coordinator.data[CONF_OPEN_ORDERS]:
+            if order["side"] == side:
+                number += 1
+
+        return number
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return len(self._get_orders())
+
+    @property
+    def unique_id(self):
+        """Return a unique id for the sensor."""
+        return self._unique_id
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return "Orders"
+
+    @property
+    def extra_state_attributes(self):
+        """Return additional sensor state attributes."""
+        return {
+            "buy orders": self._type_orders("buy"),
+            "sell orders": self._type_orders("sell"),
```

Comment:
Why are these attributes and not standalone sensors?

Reply:
Because they belong to the coin pair. and are not that important, except if you want to have a look, I have 20 different coins, this means a lot of sensors if the attributes are sensors.

Reply:
Will do a test, and see how it looks...